### PR TITLE
chore: change typing for clients

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

**Review by madlabman — APPROVED** [review]

File: src/providers/consensus/client.py [hunk 1]
```diff
@@ -192,20 +189,23 @@ def __raise_on_prysm_error(self, errors: list[Exception]) -> Exception | None:
...
-        return cast(Response, self._get(
+        data, _ = self._get(
             self.API_GET_ATTESTATION_COMMITTEES,
             path_params=(blockstamp.slot_number,),
             query_params={'epoch': epoch, 'index': index, 'slot': slot},
             stream=True
-        ))
+        )
+        if not isinstance(data, list):
+            raise ValueError("Expected list response from getEpochCommittees")  # pylint: disable=raise-missing-from
```
Comment (skhomuti) [inline src/providers/consensus/client.py:205]: this pylint rule seems to be unrelated here

**Review by skhomuti — APPROVED** [review]
